### PR TITLE
Upgrade docs

### PR DESCRIPTION
**Pull request by awholegunch**
This isn't quite finished, but there are some things that can be checked already...

This PR covers:

- the upgrade docs (and the metrics migration tool)
- the "Update table names in explore docs" (https://issues.cask.co/browse/CDAP-1671)

The last item involves updating the table name references: 
- "cdap_user_[ds-name] -> dataset_[ds-name]" and 
- "cdap_stream_[stream-name] -> stream_[stream-name]"

...everywhere...so please look and see if that looks correct.

Staged at: http://stg-web101.sw.joyent.continuuity.net/cdap/2.8.0-SNAPSHOT-r2.8_upgrade_docs/

### CODE REVIEW

File: cdap-docs/admin-manual/source/installation/installation.rst [hunk 1]
```diff
@@ -629,8 +634,20 @@ and then restart CDAP.
...
-:ref:`configuring secure Hadoop <install-secure-hadoop>`. 
+#. You should recompile and redeploy your application. This will allow you to see your old
+   run history, logs, and |---| if you migrated your old metrics with the 
+   :ref:`metric migration tool <>` |---| metrics.
+
+   **Note:** You will no longer be able to see your previous logs in the CDAP Console (UI). 
+   To access your previous logs, please see the section on downloading logs in the
+   :ref:`Logging HTTP RESTful API <http-restful-api-logging>`.
+     
+#. If you are upgrading a secure Hadoop cluster, see the notes above about
```
Comment (albertshau) [inline cdap-docs/admin-manual/source/installation/installation.rst]: not added by you, but this is redundant. they shouldn't need to see the notes if they have already set it up.

File: cdap-docs/admin-manual/source/installation/installation.rst [hunk 2]
```diff
@@ -629,8 +634,20 @@ and then restart CDAP.
...
+   :ref:`configuring secure Hadoop <install-secure-hadoop>`.
+
+   To upgrade a secure cluster, you should use ``kinit``::
```
Comment (albertshau) [inline cdap-docs/admin-manual/source/installation/installation.rst]: To upgrade CDAP on a secure Hadoop cluster, you must authenticate with kinit before running the upgrade tool

File: cdap-docs/developers-manual/source/data-exploration/custom-datasets.rst [hunk 3]
```diff
@@ -302,8 +302,8 @@ When creating your queries, keep these limitations in mind:
 - The query syntax of CDAP is a subset of the variant of SQL that was first defined by Apache Hive.
 - The SQL commands ``UPDATE`` and ``DELETE`` are not allowed on CDAP Datasets.
 - When addressing your datasets in queries, you need to prefix the data set name with the CDAP
-  namespace ``cdap_user_``. For example, if your Dataset is named ``ProductCatalog``, then the corresponding table
-  name is ``cdap_user_productcatalog``. Note that the table name is lower-case.
+  namespace ``dataset_``. For example, if your Dataset is named ``ProductCatalog``, then the corresponding table
```
Comment (albertshau) [inline cdap-docs/developers-manual/source/data-exploration/custom-datasets.rst]: how about leaving out CDAP namespace and just saying "prefix the data set name with dataset_"?  Use of "namespace" may be confusing

File: cdap-docs/developers-manual/source/data-exploration/streams.rst [hunk 4]
```diff
@@ -200,8 +200,8 @@ When creating your queries, keep these limitations in mind:
 - Writing into a Stream using SQL is not supported.
 - The SQL command ``DELETE`` is not supported.
 - When addressing your streams in queries, you need to prefix the stream name with the CDAP
-  namespace ``cdap_stream_``. For example, if your Stream is named ``Purchases``, then the corresponding table
-  name is ``cdap_stream_purchases``. Note that the table name is all lower-case, regardless of how it was defined.
+  namespace ``stream_``. For example, if your Stream is named ``Purchases``, then the corresponding table
```
Comment (albertshau) [inline cdap-docs/developers-manual/source/data-exploration/streams.rst]: similar comment as above

File: cdap-docs/admin-manual/source/installation/installation.rst [hunk 5]
```diff
@@ -601,7 +601,9 @@ and then restart CDAP.
 
 .. highlight:: console
 
-1. Stop all CDAP processes::
+1. Stop all Flows, Services, and Programs in all your applications.
```
Comment (albertshau) [inline cdap-docs/admin-manual/source/installation/installation.rst]: Flows and services are programs right? Is this supposed to say MapReduce Programs?

File: cdap-docs/admin-manual/source/installation/installation.rst (hunk 2 above)
Comment (albertshau) [inline cdap-docs/admin-manual/source/installation/installation.rst]: this note should also be further up, as you run kinit before running the upgrade tool.

File: cdap-docs/admin-manual/source/installation/installation.rst [hunk 6]
```diff
@@ -629,8 +634,20 @@ and then restart CDAP.
 
      for i in `ls /etc/init.d/ | grep cdap` ; do sudo service $i start ; done
      
-If you are upgrading a secure Hadoop cluster, see the notes above about
-:ref:`configuring secure Hadoop <install-secure-hadoop>`. 
+#. You should recompile and redeploy your application. This will allow you to see your old
```
Comment (albertshau) [inline cdap-docs/admin-manual/source/installation/installation.rst]: You must recompile and redeploy your applications

**Comment by albertshau** [comment]
a few comments

File: cdap-docs/admin-manual/source/installation/installation.rst (hunk 2 above)
Comment (awholegunch) [inline cdap-docs/admin-manual/source/installation/installation.rst]: fixed.

File: cdap-docs/admin-manual/source/installation/installation.rst (hunk 1 above)
Comment (awholegunch) [inline cdap-docs/admin-manual/source/installation/installation.rst]: fixed.

File: cdap-docs/developers-manual/source/data-exploration/custom-datasets.rst (hunk 3 above)
Comment (awholegunch) [inline cdap-docs/developers-manual/source/data-exploration/custom-datasets.rst]: fixed.

File: cdap-docs/developers-manual/source/data-exploration/streams.rst (hunk 4 above)
Comment (awholegunch) [inline cdap-docs/developers-manual/source/data-exploration/streams.rst]: fixed.

File: cdap-docs/admin-manual/source/installation/installation.rst (hunk 5 above)
Comment (awholegunch) [inline cdap-docs/admin-manual/source/installation/installation.rst]: fixed by changing to "other Programs"...

File: cdap-docs/admin-manual/source/installation/installation.rst (hunk 6 above)
Comment (awholegunch) [inline cdap-docs/admin-manual/source/installation/installation.rst]: Edited it as per comment, and moved to end as an additional step, as it didn't really follow with the other text.

**Comment by awholegunch** [comment]
Still to come: 
- "metrics migration tool" documentation
- add link in the installation doc to it
- add info on how to access old logs using Logging API: I don't think there is enough info on how to do that [at least, I couldn't do it... :) ]

**Comment by albertshau** [comment]
this part lgtm, I think we can do the points you mention in another PR.  @awholegunch do you need more info on logs?

**Comment by awholegunch** [comment]
Yes, I think we should say more on what people need to do to download the old logs that they saved as part of the Data Migration Tool. We don't tell them what to set in the [Logging RESTful API](http://stg-web101.sw.joyent.continuuity.net/cdap/2.8.0-SNAPSHOT-r2.8_upgrade_docs/en/reference-manual/http-restful-api/http-restful-api-v3/logging.html) Or do they just go off of dates?

**Comment by albertshau** [comment]
I'm not sure. @rohitsinha54 can you work with @awholegunch and make sure we have enough docs about the logs?

**Comment by albertshau** [comment]
lgtm

**Comment by rohitsinha54** [comment]
@awholegunch : We don't save them as a part of data migration. The logs are just there as they were in 2.6 and can be access in the same way through the rest api calls. 
The user just make the api calls with start and end time for their service/program or whatever they want for whatever app they have.

**Pull request merged**